### DISCUSSION TITLE
Converting SVG containing <foreignObject> text to PNG

### DISCUSSION THREAD

**Discussion by go-code** [ImageMagick/ImageMagick#6871] [Digital Image Processing] [answered]
I'm scripting a PHP conversion of some inline SVG files that contain <foreignObject> text. The text is dynamically updated based on html form input from the user (or the server). I'm using the foreignObject because it automatically centers vertically/horizontally where I need it to be, but the text isn't rendering at all in the ImageMagick PNG (or in PDF). 

Here's an example from one of the svg files: `<foreignObject x="57.087" y="67.308" width="355.8261" height="55.3115" class="homeTownName"><div xmlns="http://www.w3.org/1999/xhtml" style="height: 55.3115px; font-size: 22px; font-weight: 400; text-transform: uppercase; color: rgb(255, 0, 25); display: flex; text-align: center; align-items: center; justify-content: center; font-family: verveine, sans-serif; font-style: normal;">Philadelphia</div></foreignObject> `

I'd appreciate any suggestions. Thanks.

**Comment by snibgo** [comment, marked as answer]
foreignObject isn't (yet) part of standard SVG, and is not (yet) supported by InkScape, or by SVG renderers that are internal to IM.
